### ISSUE TITLE
[SkullUtils] - GameProfile

### ISSUE THREAD

**Issue by divios** [CryptoMorin/XSeries#169]
**Description**
Hi! I recently noticed that setting a PLAYER_HEAD skull value from base64 with the method SkullUtils#applySkin (not checked if with the other methods happens) causes a NullPointerException when comparing that item with ItemStack#isSimilar or ItemStack#equals.

```java.lang.NullPointerException: Cannot invoke "net.minecraft.nbt.NBTTagCompound.equals(Object)" because "this.serializedProfile" is null
[14:55:24 WARN]: 	at org.bukkit.craftbukkit.v1_18_R2.inventory.CraftMetaSkull.equalsCommon(CraftMetaSkull.java:247)
[14:55:24 WARN]: 	at org.bukkit.craftbukkit.v1_18_R2.inventory.CraftItemFactory.equals(CraftItemFactory.java:316)
[14:55:24 WARN]: 	at org.bukkit.craftbukkit.v1_18_R2.inventory.CraftItemFactory.equals(CraftItemFactory.java:303)
[14:55:24 WARN]: 	at org.bukkit.inventory.ItemStack.isSimilar(ItemStack.java:294)
```

With a quick google search I found someone already faced a similar bug, this was his reply: 

> Modified ItemFactory#createSkull and TexturedSkullFactory#createSkull so they use CraftMetaSkull#setProfile(GameProfile) when available rather than setting the field directly. This fixes it because a change in 1.15 requires a new field, serializedProfile, to be set and checked. Setting the profile field directly doesn't set this serializedProfile field, and it throws an error

**Version**
1.18.1

**Comment by CryptoMorin** [comment]
If you can make a PR for it using Java's Method Handle API, go ahead. If not, that's fine too thanks for sharing the solution!

**Comment by CryptoMorin** [comment]
I just wanted to leave a reproducible code here since this doesn't happen to all items. It only happens when comparing two skulls that **have a custom texture set.**

```java
ItemStack item = XMaterial.PLAYER_HEAD.parseItem();
ItemMeta meta = item.getItemMeta();
SkullUtils.applySkin(meta, "eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvYzVkYjU3MzczMDkwNDA0MWRlZTkzN2JlZWFkYThiMDE4MjM1MTkyNGE0MmY5OTc5NTEzY2YwYmQ0MzNiZTA3NyJ9fX0=");
item.setItemMeta(meta);

ItemStack secItem = XMaterial.PLAYER_HEAD.parseItem();
ItemMeta secMeta = secItem.getItemMeta();
SkullUtils.applySkin(secMeta, "eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvNGI4NWJmMWY1YjY1MWMwMTIwN2JlN2VjYWEzNmNiYWZhOGUxZDJhOWE0YTVkNTNjNTg3NWI0ODkyNjZkNjYxOSJ9fX0=");
secItem.setItemMeta(secMeta);

secItem.isSimilar(item);
```

**Comment by CryptoMorin** [comment]
Fixed in v8.7.1
